kernel/src/syscall.rs

```diff
@@ -394,12 +395,40 @@ pub enum SyscallReturnVariant {
     FailureU32 = 1,
     FailureU32U32 = 2,
     FailureU64 = 3,
+    FailurePtrUsize = 4,
+    FailurePtrPtr = 5,
     Success = 128,
     SuccessU32 = 129,
     SuccessU32U32 = 130,
     SuccessU64 = 131,
     SuccessU32U32U32 = 132,
     SuccessU32U64 = 133,
+    SuccessUsize = 134,
+    SuccessPtr = 135,
+    SuccessPtrUsize = 136,
+    SuccessPtrPtr = 137,
+}
+
+impl SyscallReturnVariant {
+    /// Maps newly introduced return variants (usize and ptr)
+    /// to old ones (u32) for backwards compatibility.
+    /// This should not be used for any newly designed interfaces,
+    /// and will eventually be deprecated once all interfaces are updated.
+    pub const fn into_compat(self) -> Self {
```

Comment:
Maybe also rename this to something like `into_compat_32bit_trd104` or something else meaningfully ugly?

Reply:
Yeah, this should be TRD 104-specific. It most likely should also go into the external module that I propose in #4229. It's defines the architecture-specific encoding of syscall returns, whereas the enum type itself is what we use as the canonical representation on the core kernel.

Reply:
I moved this type to another module in #4229. We can either merge this one first and then rename it in this PR, or rename it here.

In fact, I'd argue for not making this an associated method at all, but rather a private function in the `encode_syscall_return` function. It's an implementation detail.

Reply:
Renamed as suggested